### PR TITLE
[1772, 1796] Authenticate calls to /delegates route

### PR DESCRIPTION
**Pull request by Flip-Liquid**
## PR-Codex overview
This PR focuses on enhancing API user authentication and key generation. It includes middleware updates, API key generation script, and Prisma schema changes.

### Detailed summary
- Added new middleware for API key validation
- Implemented API key generation script
- Updated Prisma schema for API users
- Modified authentication logic for API users
- Updated dependencies including `uuid` and `ts-node`

> The following files were skipped due to too many changes: `yarn.lock`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/app/lib/middleware/auth.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  }
+
+  let authResponse: AuthResponse = hasApiKey(request);
+
+  const token = request.headers.get("authorization");
+
+  // TODO: caching logic, rate limiting
+  const user = await prisma.api_user.findFirst({
+    where: {
+      api_key: token,
```
Comment (stepandel) [inline src/app/lib/middleware/auth.ts:49]: Best practice is to hash these

**Review by stepandel — COMMENTED** [review]
If we're storing API keys in the DB, we should be hashing them. But considering the middleware constrain (no prisma in edge runtime), wondering if using an encryption based key like JWT might be a better approach there?

File: src/app/lib/middleware/auth.ts (hunk 1 above)
Comment (Flip-Liquid) [reply src/app/lib/middleware/auth.ts:49]: true true, miss on my part

Comment (Flip-Liquid) [reply src/app/lib/middleware/auth.ts]: Addressed this, we're now comparing the hash

**Review by stepandel — APPROVED** [review]
LGTM!

**Pull request merged**